From ae342c47c31eca866ec35960fbecaf4da7d33b1b Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Thu, 12 Mar 2020 13:30:39 -0700
Subject: [PATCH 1284/1852] v4l: xilinx: vipp: Change xvip_entity_start/stop to
 take graph entity

commit 4c28f6b0c7bcd8f54b5ed2b283c2854d4b58324f from
https://github.com/Xilinx/linux-xlnx.git

Direct use of xvip_graph_entity will allow translation from subdev back
to graph entity.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Dylan Yip <dylan.yip@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vipp.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vipp.c b/drivers/media/platform/xilinx/xilinx-vipp.c
index 66559c8dd04e..1d53cdeee009 100644
--- a/drivers/media/platform/xilinx/xilinx-vipp.c
+++ b/drivers/media/platform/xilinx/xilinx-vipp.c
@@ -240,15 +240,16 @@ static bool xvip_subdev_set_streaming(struct xvip_composite_device *xdev,
 	return false;
 }
 
-static int xvip_entity_start_stop(struct xvip_composite_device *xdev,
-				  struct media_entity *entity, bool on)
+static int
+xvip_graph_entity_start_stop_subdev(struct xvip_composite_device *xdev,
+				    struct xvip_graph_entity *entity, bool on)
 {
 	struct v4l2_subdev *subdev;
 	int ret = 0;
 
 	dev_dbg(xdev->dev, "%s entity %s\n",
-		on ? "Starting" : "Stopping", entity->name);
-	subdev = media_entity_to_v4l2_subdev(entity);
+		on ? "Starting" : "Stopping", entity->entity->name);
+	subdev = media_entity_to_v4l2_subdev(entity->entity);
 
 	/*
 	 * start or stop the subdev only once in case if they are
@@ -357,7 +358,7 @@ static bool xvip_graph_entity_start_stop(struct xvip_composite_device *xdev,
 		return true;
 	}
 
-	ret = xvip_entity_start_stop(xdev, entity->entity, on);
+	ret = xvip_graph_entity_start_stop_subdev(xdev, entity, on);
 	if (ret < 0) {
 		dev_err(xdev->dev, "ret = %d for entity %s\n",
 			ret, entity->entity->name);
-- 
2.31.1

